hardhat/contracts/AirChrysalis.sol

```diff
@@ -54,74 +51,62 @@ contract AirChrysalis is VRFConsumerBaseV2 {
     }
 
     function createGiveaway(address _contractAddr, uint256[2][] memory _tokensTuple) public {
-        for (uint256 idx = 0; idx < _tokensTuple.length; idx++) {
-            ERC1155 token = ERC1155(_contractAddr);
-            require(token.balanceOf(msg.sender, _tokensTuple[idx][0]) > 0, "caller must own given token");
-            require(token.balanceOf(msg.sender, _tokensTuple[idx][0]) >= _tokensTuple[idx][1], "caller must have specified amount of token");
-        }
-        addressGiveaways[msg.sender].push(giveawayId);
         address[] memory participants;
-        giveawaysMap[msg.sender][giveawayId] = Giveaway(_contractAddr, _tokensTuple, participants, false, address(0), false);
-        emit GiveawayCreated(msg.sender, giveawayId);
-        giveawayId++;
+        giveaways.push(Giveaway(_contractAddr, _tokensTuple, participants, false, address(0), msg.sender, 0));
+        addressGiveaways[msg.sender].push(giveaways.length);
+        emit GiveawayCreated(msg.sender, giveaways.length);
```

Comment:
Do we need a global place to track giveaways? 

I would remove `giveaways` and just do 

```ts
addressGiveaways[msg.sender].push(Giveaway(_contractAddr, _tokensTuple, participants, false, address(0), msg.sender, 0));
emit GiveawayCreated(msg.sender, addressGiveaways[msg.sender].length);
```

This would save a bit more than 20k gas